tests/ci/cdk/cdk/bm_framework_stack.py

```diff
@@ -0,0 +1,147 @@
+                                      iam.ManagedPolicy.from_aws_managed_policy_name("CloudWatchAgentServerPolicy")
+                                  ])
+
+        # Create build spec.
+        placeholder_map = {"ECR_REPO_PLACEHOLDER": ecr_arn(ecr_repo_name)}
+        build_spec_content = YmlLoader.load(spec_file_path, placeholder_map)
+
+        # Define CodeBuild.
+        project = codebuild.Project(
+            scope=self,
+            id=id,
+            project_name=id,
+            source=git_hub_source,
+            role=codebuild_role,
+            timeout=core.Duration.minutes(180),
+            environment=codebuild.BuildEnvironment(compute_type=codebuild.ComputeType.SMALL,
+                                                   privileged=False,
+                                                   build_image=codebuild.LinuxBuildImage.STANDARD_4_0),
+            build_spec=codebuild.BuildSpec.from_object(build_spec_content))
+
+        # Add 'BuildBatchConfig' property, which is not supported in CDK.
+        # CDK raw overrides: https://docs.aws.amazon.com/cdk/latest/guide/cfn_layer.html#cfn_layer_raw
+        # https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-codebuild-project.html#aws-resource-codebuild-project-properties
+        cfn_build = project.node.default_child
+        cfn_build.add_override("Properties.BuildBatchConfig", {
+            "ServiceRole": codebuild_role.role_arn,
+            "TimeoutInMins": 180
+        })
+
+        # create iam for ec2s
+        s3_read_write_policy_prod_bucket = iam.PolicyDocument.from_json(s3_read_write_policy_in_json(S3_PROD_BUCKET))
+        s3_read_write_policy_pr_bucket = iam.PolicyDocument.from_json(s3_read_write_policy_in_json(S3_PR_BUCKET))
+        ec2_inline_policies = {"s3_read_write_policy_prod_bucket": s3_read_write_policy_prod_bucket,
+                               "s3_read_write_policy_pr_bucket": s3_read_write_policy_pr_bucket}
+        ec2_role = iam.Role(scope=self, id="{}-ec2-role".format(id),
+                            role_name="{}-ec2-role".format(id),
+                            assumed_by=iam.ServicePrincipal("ec2.amazonaws.com"),
+                            inline_policies=ec2_inline_policies,
+                            managed_policies=[
+                                iam.ManagedPolicy.from_aws_managed_policy_name("AmazonSSMManagedInstanceCore"),
+                                iam.ManagedPolicy.from_aws_managed_policy_name("CloudWatchAgentServerPolicy")
+                            ])
+        iam.CfnInstanceProfile(scope=self, id="{}-ec2-profile".format(id),
+                               roles=["{}-ec2-role".format(id)],
+                               instance_profile_name="{}-ec2-profile".format(id))
+
+        # create vpc for ec2s
+        vpc = ec2.Vpc(self, id="{}-ec2-vpc".format(id))
+
+        # create security group with default rules
+        ec2.SecurityGroup(self, id="{}-ec2-sg".format(id),
```

Comment:
Will `destroy-ci` clean up these resources(`s3`, `cloudwatch logs` and `security group`)?

Reply:
I'm unsure that we want `destroy-ci` to delete the s3 buckets, as that will wipe our benchmarking history should we need to delete our ci somehow. I will look into deleteing both `cloudwatch logs` and `security group` when destroy-ci is run though 👍

Reply:
In team's account, the history should be protected. But for individual account, we want to clean up these resources when not needed.

Reply:
This is true, I will look into having `destroy-ci` delete the s3 bucket if it is not being run on the team's account.